task/control-plane-agent/src/mgs_common.rs

```diff
@@ -32,6 +33,9 @@ impl MgsCommon {
     pub(crate) fn claim_static_resources(base_mac_address: MacAddress) -> Self {
         Self {
             reset_requested: false,
+            reset_component_requested: SpComponent {
+                id: [0; SpComponent::MAX_ID_LENGTH],
```

Comment:
This looks like it's intended to be an invalid SpComponent, which suggests that Option is more appropriate.